jnius/reflect.py

```diff
@@ -335,3 +323,31 @@ def _getitem(self, index):
         clsname,
         (JavaClass, ),
         classDict)
+
+
+## dunder method for List
+def _getitem(self, index):
+    try:
+        return self.get(index)
+    except JavaException as e:
+        # initialize the subclass before getting the Class.forName
+        # otherwise isInstance does not know of the subclass
+        mock_exception_object = autoclass(e.classname)()
+        if find_javaclass("java.lang.IndexOutOfBoundsException").isInstance(mock_exception_object):
+            # python for...in iteration checks for end of list by waiting for IndexError
+            raise IndexError()
+        else:
+            raise
+
+# protocol_map is a user-accessible API for patching class instances with additional methods 
+protocol_map = {
+    'java.util.List' : {
+        '__getitem__' : _getitem,
+        '__len__' : lambda self: self.size()
```

Comment:
I think `__len__` -> .size() can actually be applied to the super interface `java.util.Collection`.